src/main/java/com/google/devtools/build/lib/bazel/bzlmod/RegistryFactoryImpl.java

```diff
@@ -25,6 +27,7 @@
 public class RegistryFactoryImpl implements RegistryFactory {
   private final DownloadManager downloadManager;
   private final Supplier<Map<String, String>> clientEnvironmentSupplier;
+  private final Cache<String, Registry> registries = Caffeine.newBuilder().build();
```

Comment:
Should I make this a plain `ComcurrentHashMap`? I guess we don't need fancy properties of the cache, especially since we use strong keys and values.

Reply:
I don't really mind. This is simple enough to read that I don't think it matters that much.